### PR TITLE
Create new post endpoint in dynamics adapter to receive match found events

### PR DESCRIPTION
**Pull request by peggy-gov-bc-ca**
# Description

Create new post endpoint in dynamics adapter to receive match found events.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested with Postman to post MatchFoundResponse to this endpoint.

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

this PR fixes jira ticket: **FAMS-599**

### CODE REVIEW

File: app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+namespace DynamicsAdapter.Web.MatchFound
+{
+    public class MatchFoundResponse
+    { 
+        public string SearchRequestId { get; set; }
+        public string SearchResult { get; set; }
+    }
+
+
+    [Route("api/[controller]")]
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: can we go just `"[controller]"` dotnet core 3 is getting rid of the api path

File: app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    {
+        private readonly ILogger<MatchFoundController> _logger;
+        public MatchFoundController(ILogger<MatchFoundController> logger)
+        {
+            _logger = logger;
+        }
+
+        //POST
+        [HttpPost]
+        public async Task<IActionResult> MatchFound(MatchFoundResponse response)
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: what we want here is the id in the path like `MatchFound/{id}`, then your method can take (Guid id, string payload), you have to use the `route` attribute to define the id. can you also add the `[Produces]`, `[ProducesResponseType]` attributes

File: app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs (hunk 1 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: sure

File: app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs (hunk 2 above)
Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: sure

File: app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        [ProducesResponseType(StatusCodes.Status400BadRequest)]
+        public async Task<IActionResult> MatchFound(string id, [FromBody]Object payload)
+        {
+            if (string.IsNullOrEmpty(id))
+            {
+                return BadRequest();
+            }
+
+            _logger.LogInformation("Received MatchFound response with SearchRequestId is " + id);
+            _logger.LogInformation("SearchResult is " + payload);
```
Comment (alexjoybc) [inline app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: just remove that line, we don't want to log any PI info :-)

Comment (peggy-gov-bc-ca) [reply app/DynamicsAdapter/DynamicsAdapter.Web/MatchFound/MatchFoundController.cs]: ok

**Pull request merged**